### DISCUSSION TITLE
viewTypes in package.json vs registerCustomEditorProvider()

### DISCUSSION THREAD

**Discussion by rileydanejohnston** [microsoft/vscode-discussions#512] [Extension Development QnA]
Hello,
My understanding is that the viewType specified in the package.json should match the viewType passed to registerCustomEditorProvider(). Currently, I have different viewType values in each and my extension runs just fine. When I update the viewType passed to registerCustomEditorProvider() to match the package.json (or a different random value), I get a continuous loading screen with no errors in the debug console or devTools view. Am I missing something here? [The docs](url) seem to indicate these values should match.

package.json:
```
"customEditors": [
      {
        "viewType": "diagram.editor",
        "displayName": "Diagram Editor",
        "selector": [ ... ],
        "priority": "option"
      }
    ],
```
extension.ts:
```
const customEditorProvider = vscode.window.registerCustomEditorProvider(
        'diagram.editor',
        new DiagramProvider(context, vscodeConnector),
        {
            webviewOptions: { retainContextWhenHidden: true },
            supportsMultipleEditorsPerDocument: true
        }
    );
```
result:

https://user-images.githubusercontent.com/54558543/223601508-3c5d138c-3dcb-4023-b02f-fb6ecae3039a.mov

**Comment by isidorn** [comment]
I am not sure. Are you still hitting this issue? If yes I can reach out to some of the devs for help :)

**Comment by burkeholland** [comment]
Closing because this is more than 30 days old. If you would like us to reach out to the devs as @Isidorn mentions above, please re-open and let us know.
